### DISCUSSION TITLE
Receive data if device connected or disconnected to ESP32.

### DISCUSSION THREAD

**Discussion by Oppa-YA** [pschatzmann/ESP32-A2DP#115] [Q&A] [answered]
Hello, just tested your library, it's cool and sounds nice with PCM5102. 
I have a question. Could I receive info if any device connected or disconnected to ESP32 receiver?
I want to use it like that: If device connect to receiver i wanna send IR code to switch my speakers input to one which is connected to the ESP32 receiver, and also when i disconnect from it i want to make it send another code to IR and switch speakers back to input where is my PC connected.
It would be very useful and i think many people would be interested. 
Could you please suggest if this possible to do with last build or if no, or could you please add this possibility? 
Thank you!

**Comment by pschatzmann** [comment, marked as answer]
I am not sure if I understand you correctly: but have you looked at the example: bt_music_receiver_status_callback
and the documentation ?

https://pschatzmann.github.io/ESP32-A2DP/html/annotated.html

**Reply by Oppa-YA** [reply]
![Screenshot 2021-11-02 134739](https://user-images.githubusercontent.com/11776599/139840683-161ffa19-d8dc-4e61-9540-f3ea766e4d13.png)
I did not see "bt_music_receiver_status_callback" example, I found only example file "bt_music_receiver_datacallback". Does you talk about it?
I mean i want to read status of the connection and write something a like that code:
_if status (ESP_A2D_CONNECTION_STATE_CONNECTED) 
{send ir code ON}_

_if  status (ESP_A2D_CONNECTION_STATE_DISCONNECTED) 
{send ir code OFF}_

I checked arduino console after uploading "bt_music_receiver_32bit_ext" but i only see status like DISCONNECTED in the console, and it never shows CONNECTED.

I also tried flashing bt_music_receiver_datacallback example but it dont work for me, only showing zeros in the console.
Can you please suggest what to do next?

**Comment by pschatzmann** [comment]
https://github.com/pschatzmann/ESP32-A2DP/blob/main/examples/bt_music_receiver_status_callback/bt_music_receiver_status_callback.ino

**Reply by Oppa-YA** [reply]
Thank you. Could you suggest how to read status and use it in code?
I tried many variants but cant get it to work.

![Screenshot 2021-11-02 150617](https://user-images.githubusercontent.com/11776599/139853160-658439c7-9a74-43b8-8ce5-e610ce03a6f0.png)

**Reply by pschatzmann** [reply]
If you plan to use it in the loop anyhow why don't you call a2dp_sink.is_connected() ?
in the callback you get more information 

```
void connection_state_changed(esp_a2d_connection_state_t state, void *ptr){
  if (state==ESP_A2D_CONNECTION_STATE_CONNECTED){
     // do action on connect
  }
}
```

**Reply by Oppa-YA** [reply]
Much thanks you for suggestions and help! I can't get that exact code (last one you posted) to compile (maybe i just put in wrong place) but after i took some code from  "bt_music_receiver_with_metadata" I finally got working code like that, after connect LED is light up. Now that main part is working i will try to implement IR code sender instead of LED:


```
#include "BluetoothA2DPSink32.h"
BluetoothA2DPSink32 a2dp_sink; // Subclass of BluetoothA2DPSink
const byte led_gpio = 32;

void setup() {
  a2dp_sink.set_bits_per_sample(32);  
  a2dp_sink.start("Hifi32bit");  
  pinMode(led_gpio, OUTPUT);
}

void loop() {
  if (a2dp_sink.get_connection_state()==ESP_A2D_CONNECTION_STATE_CONNECTED){
  digitalWrite (led_gpio, HIGH);}

  else if(a2dp_sink.get_connection_state()==ESP_A2D_CONNECTION_STATE_DISCONNECTED){
  digitalWrite (led_gpio, LOW);}
}
```
